src/libraries/System.Security.Cryptography.Pkcs/src/Resources/Strings.resx

```diff
@@ -283,6 +283,9 @@
   <data name="Cryptography_Cms_InvalidSymmetricKey" xml:space="preserve">
     <value>The key in the enveloped message is not valid or could not be decoded.</value>
   </data>
+  <data name="Cryptography_Cms_DigestAsSignatureNotSupported" xml:space="preserve">
+    <value>Using a digest as the signature is not supported for the specified signature algorithm.</value>
+  </data>
   <data name="Cryptography_X509_PfxWithoutPassword_MaxAllowedIterationsExceeded" xml:space="preserve">
```

Comment:
I think this is describing the technical aspect fine... but no one would understand it.

The message seems more like it should be `SignatureIdentifierType.NoSignature is not valid with the provided certificate.`  Or `SignatureIdentifierType.NoSignature is not valid with public key algorithm {0}` (the Subject Public Key Info algorithm, not the certificate signer's signature algorithm).

Presumably the resource name would change at that point, but it might still be fine with that rewritten message.  The important point is, the user didn't ask for "using a digest as the signature"... the thing they did (and need to change) is using the NoSignature value.